File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/effect/UXColorAdjustEffect.php
<?php
namespace php\gui\effect;

use php\gui\paint\UXColor;

/**
 * Class UXColorAdjustEffect
 * @package php\gui\effect
 * @packages gui, javafx
 */
class UXColorAdjustEffect extends UXEffect
{
    /**
     * @var double   -1.0 to 1.0
     */
    public $brightness = 0.0;

    /**
     * @var double   -1.0 to 1.0
     */
    public $contrast = 0.0;

    /**
     * @var double   -1.0 to 1.0
     */
    public $hue = 0.0;

    /**
     * @var double   -1.0 to 1.0
     */
    public $saturation = 0.0;
}

File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/effect/UXColorAdjustEffect.java
package org.develnext.jphp.ext.javafx.classes.effect;

import javafx.scene.effect.Bloom;
import javafx.scene.effect.ColorAdjust;
import javafx.scene.effect.Effect;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Name(JavaFXExtension.NS + "effect\\UXColorAdjustEffect")
public class UXColorAdjustEffect extends UXEffect<ColorAdjust> {
    interface WrappedInterface {
        @Property double brightness();
        @Property double contrast();
        @Property double hue();
        @Property double saturation();

        @Property Effect input();
    }

    public UXColorAdjustEffect(Environment env, ColorAdjust wrappedObject) {
        super(env, wrappedObject);
    }

    public UXColorAdjustEffect(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct() {
        __wrappedObject = new ColorAdjust();
    }
}
